walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -14,10 +14,76 @@
 
 package com.google.sps;
 
-import java.util.Collection;
+import java.util.*;
 
 public final class FindMeetingQuery {
   public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
+    int duration = (int)request.getDuration();
+
+    // If request is longer than length of day, then there would never be any options
+    if (duration > TimeRange.WHOLE_DAY.duration()) {
+      return Arrays.asList();
+    }
+
+    // Seperate relevant TimeRanges from events, put into ArrayList and sort by ascending meeting start time
+    ArrayList<TimeRange> meetings = new ArrayList<TimeRange>();
+    events.forEach(event -> {
+      // First check if the event in question contains people from request
+      Set<String> attendees = new HashSet<String>(request.getAttendees());
+      attendees.retainAll(event.getAttendees());
+
+      int numMeetings = meetings.size();
+
+      // Only add meeting if A) there are people attending and B) it's not nested within the previous meeting
+      if(!attendees.isEmpty() && 
+        (numMeetings == 0 || event.getWhen().end() > meetings.get(numMeetings - 1).end())) {
```

Comment:
How does this check the nested meeting? Are the meetings already sorted by starts (and ends)?

Reply:
meetings is sorted by start time, so if an event with a later start time also has an earlier end time, it's not included.

Reply:
Don't you sort the meetings outside this loop? Is the "event" in events sorted?

Reply:
Huh, you're right. I guess the test cases always gave a well-ordered list of events! Thanks for catching, fixed in latest commit.